src/core/client/ui/components/v2/Timestamp/Timestamp.tsx

```diff
@@ -12,31 +10,22 @@ export interface TimestampProps {
   onToggleAbsolute?: (showAbsolute: boolean) => void;
 }
 
-const Timestamp: FunctionComponent<TimestampProps> = (props) => {
-  const [showAbsolute, setShowAbsolute] = useState(false);
-  const toggleShowAbsolute = useCallback(() => {
-    if (props.toggleAbsolute) {
-      if (props.onToggleAbsolute) {
-        props.onToggleAbsolute(!showAbsolute);
-      }
-
-      setShowAbsolute(!showAbsolute);
-    }
-  }, [showAbsolute, setShowAbsolute, props]);
+const Timestamp: FunctionComponent<TimestampProps> = ({
+  className,
+  children,
+  toggleAbsolute,
+  onToggleAbsolute,
+}) => {
   return (
-    <BaseButton className={styles.root} onClick={toggleShowAbsolute}>
-      {showAbsolute ? (
-        <AbsoluteTime
-          date={props.children}
-          className={cn(styles.text, props.className)}
-        />
-      ) : (
-        <RelativeTime
-          className={cn(styles.text, props.className)}
-          date={props.children}
-        />
-      )}
-    </BaseButton>
+    <TimestampFormatter
+      className={className}
+      rootClassName={styles.root}
+      textClassName={styles.text}
+      toggleAbsolute={toggleAbsolute}
+      onToggleAbsolute={onToggleAbsolute}
+    >
+      {children}
+    </TimestampFormatter>
```

Comment:
Lack of css styling that makes it bolded/off-grey

Reply:
I see! I think the name is misleading though, because usually the name of the component suggest what will be rendered. In this case I would expect an interactive timestamp formatter UI to be rendered. It's not perfect but maybe we can do `<Timestamp raw />`, use the `variant` prop, or something else.

Reply:
Fair enough, I think I'll just remove both of these abstractions and override the classname instead